core/src/main/java/com/sequenceiq/cloudbreak/service/stack/GatewaySecurityGroupWithDefaultGatewayCidrsDecorator.java

```diff
@@ -0,0 +1,91 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import static com.sequenceiq.cloudbreak.service.securityrule.SecurityRuleService.TCP_PROTOCOL;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import com.sequenceiq.cloudbreak.domain.SecurityGroup;
+import com.sequenceiq.cloudbreak.domain.SecurityRule;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceGroup;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+import com.sequenceiq.common.api.type.Tunnel;
+
+@Component
+public class GatewaySecurityGroupWithDefaultGatewayCidrsDecorator {
```

Comment:
Can we use a simpler name? Like GatewaySecurityGroupDecorator?

Reply:
renamed